src/main/java/org/opentripplanner/routing/algorithm/AStar.java

```diff
@@ -95,8 +95,15 @@ public ShortestPathTree getShortestPathTree(RoutingRequest req, double relTimeou
         return this.getShortestPathTree(req, relTimeoutSeconds, null);
     }
     
+    /** set up a single-origin search */
     public void startSearch(RoutingRequest options,
             SearchTerminationStrategy terminationStrategy, long abortTime) {
+        startSearch(options, terminationStrategy, abortTime, true);
+    }
+    
+    /** set up the search, optionally not adding the initial state to the queue (for multi-state Dijkstra) */
+    private void startSearch(RoutingRequest options,
```

Comment:
I'm wondering if we should always use the same process no matter how many initial states, e.g. caller always constructs and inserts the initial states.